test/audits/pdp.test.js

```diff
@@ -0,0 +1,282 @@
+
+  beforeEach(() => {
+    context = new MockContextBuilder()
+      .withSandbox(sandbox)
+      .withOverrides({
+        log: {
+          info: sinon.stub(),
+          warn: sinon.stub(),
+          error: sinon.stub(),
+        },
+      })
+      .build(message);
+
+    googleClientStub = {
+      urlInspect: sandbox.stub(),
+    };
+
+    urlInspectStub = googleClientStub.urlInspect;
+    siteStub = {
+      getId: () => '123',
+      getConfig: () => ({
+        getProductDetailPages: () => ['https://example.com/product/1', 'https://example.com/product/2'],
+      }),
+    };
+
+    fullUrlInspectionResult = {
+      inspectionResult: {
+        inspectionResultLink: 'https://search.google.com/search-console/inspect?resource_id=https://www.example.com/',
+        indexStatusResult: {
+          verdict: 'PASS',
+          coverageState: 'Submitted and indexed',
+          robotsTxtState: 'ALLOWED',
+          indexingState: 'INDEXING_ALLOWED',
+          lastCrawlTime: '2024-08-13T22:35:22Z',
+          pageFetchState: 'SUCCESSFUL',
+          googleCanonical: 'https://www.example.com/foo',
+          userCanonical: 'https://www.example.com/foo',
+          referringUrls: [
+            'https://www.example.com/bar',
+          ],
+          crawledAs: 'MOBILE',
+        },
+        mobileUsabilityResult: {
+          verdict: 'VERDICT_UNSPECIFIED',
+        },
+        richResultsResult: {
+          verdict: 'PASS',
+          detectedItems: [
+            {
+              richResultType: 'Product snippets',
+              items: [
+                {
+                  name: 'Example Product Name',
+                  issues: [
+                    {
+                      issueMessage: 'Missing field "image"',
+                      severity: 'ERROR',
+                    },
+                  ],
+                },
+              ],
+            },
+            {
+              richResultType: 'Merchant listings',
+              items: [
+                {
+                  name: 'Example Product Name',
+                  issues: [
+                    {
+                      issueMessage: 'Missing field "hasMerchantReturnPolicy"',
+                      severity: 'WARNING',
+                    },
+                    {
+                      issueMessage: 'Missing field "shippingDetails"',
+                      severity: 'ERROR',
+                    },
+                  ],
+                },
+              ],
+            },
+          ],
+        },
+      },
+    };
+  });
+
+  afterEach(() => {
+    sandbox.restore();
+  });
+
+  it('should successfully return a filtered result of the url inspection result', async () => {
+    sandbox.stub(GoogleClient, 'createFrom').returns(googleClientStub);
+    urlInspectStub.resolves(fullUrlInspectionResult);
+
+    const auditData = await gscPdpStructuredDataHandler('https://www.example.com', context, siteStub);
+
+    expect(auditData.auditResult).to.deep.equal(
```

Comment:
While this expect is somewhat less prone to need updates in case of fullUrlInspectionResult being updated, it doesn't work as documentation on what the audit result looks like.
If we treat unit tests as developer documentation, then having here the explicit JSON would be more helpful.